src/snapshot/src/lib.rs

```diff
@@ -340,6 +340,130 @@ mod tests {
         }
     }
 
+    #[cfg(target_arch = "x86_64")]
+    #[test]
+    fn test_get_format_version_x86() {
+        // Check if `get_format_version()` returns indeed the format
+        // version (the least significant 2 bytes) if the id is valid
+        // (the other bytes == BASE_MAGIC_ID).
+        let mut magic_id = 0x0710_1984_8664_0001u64;
+        assert_eq!(get_format_version(magic_id).unwrap(), 1u16);
+        magic_id = 0x0710_1984_8664_000Eu64;
+        assert_eq!(get_format_version(magic_id).unwrap(), 14u16);
+
+        // Flip a bit to invalidate the arch id.
+        magic_id = 0x1710_1984_8664_0001u64;
+        assert_eq!(
+            get_format_version(magic_id).unwrap_err(),
+            Error::InvalidMagic(magic_id)
+        );
+    }
+
+    #[cfg(target_arch = "aarch64")]
+    #[test]
+    fn test_get_format_version_aarch64() {
+        let mut magic_id = 0x0710_1984_AAAA_0001u64;
+        assert_eq!(get_format_version(magic_id).unwrap(), 1u16);
+        magic_id = 0x0710_1984_AAAA_000Eu64;
+        assert_eq!(get_format_version(magic_id).unwrap(), 14u16);
+        magic_id = 0x1710_1984_AAAA_0001u64;
+        assert_eq!(
+            get_format_version(magic_id).unwrap_err(),
+            Error::InvalidMagic(magic_id)
+        );
+    }
```

Comment:
Can we test the `get_format_version` in a single test where we define `magic_id` per architecture? We can use the same value for the `Error::InvalidMagic` case. The code for both tests is pretty much the same if you disregard the magic number.

Reply:
Updated it.